pyproject.toml

```diff
@@ -3,10 +3,45 @@ requires = [
     "setuptools>=46.4.0",       # PEP-420 support, PEP-517/518 support, setup.cfg attr: support
     "wheel>=0.30.0",            # limited python api support
     "cython~=3.0",
-    'oldest-supported-numpy',
+    "numpy==1.24.4; python_version=='3.8'",
+    "numpy>=2.0.0rc1; python_version>='3.9'",
 ]
 build-backend = "setuptools.build_meta"
 
+[project]
```

Comment:
I think it's fine either way, but out of curiosity why get rid of `setup.cfg`?

Reply:
I do see that [cibuildwheel](https://github.com/pypa/cibuildwheel) which is maintained by pypa has ditched their `setup.cfg`, so I am guessing that's what they want us to do. It's been quite a moving target over the last few years.

Reply:
Sure, using only `pyproject.toml` just seems to be the recommended way (see [this ](https://packaging.python.org/en/latest/guides/modernize-setup-py-project/)and [this](https://peps.python.org/pep-0621/)). I also think it's nicer keeping it simple, rather than having everything separated into several files.

Reply:
Sigh. Ok, I find [this](https://packaging.python.org/en/latest/guides/modernize-setup-py-project/) pretty definitive. I do wish they would pick a format and stick to it.